pkg/blobserver/diskpacked/diskpacked.go

```diff
@@ -125,9 +125,26 @@ func IsDir(dir string) (bool, error) {
 // files.
 func New(dir string) (blobserver.Storage, error) {
 	var maxSize int64
-	if ok, _ := IsDir(dir); ok {
-		// TODO: detect existing max size from size of files, if obvious,
-		// and set maxSize to that?
+	var n, atMax int
+	if dh, err := os.Open(dir); err == nil {
+		if fis, err := dh.Readdir(-1); err == nil {
+			// Detect existing max size from size of files, if obvious, and set maxSize to that
+			for _, fi := range fis {
+				if nm := fi.Name(); strings.HasPrefix(nm, "pack-") && strings.HasSuffix(nm, ".blobs") {
+					n++
+					if s := fi.Size(); s > maxSize {
+						maxSize, atMax = fi.Size(), 0
+					} else if s == maxSize {
+						atMax++
+					}
+				}
+			}
+		}
+	}
+	// Believe to the deduced size only if at least 2 files has that maximum size,
+	// and all files (except one) has the same.
+	if !(atMax > 1 && n == atMax+1) {
```

Comment:
This check can be moved inside the `if` statement. We only need to do this if we have looped over all files. So it can be moved directly after the for loop, inside the same statement.